atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/asts/bucket/SweepBucketStore.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.sweep.asts.bucket;
+
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Stores information about the ranges of timestamps associated with given bucket identifiers.
+ */
+public interface SweepBucketStore {
+    /**
+     * Retrieves the range of timestamps associated with the given bucket identifier, or an empty optional if the
+     * bucket does not exist.
+     */
+    Optional<SweepableBucketRange> getBucketRange(long bucketIdentifier);
+
+    /**
+     * Retrieves a list of buckets corresponding to the lowest buckets the sweep bucket store knows about, that are
+     * at least as high as the provided AtlasDB timestamp. This list is guaranteed to be sorted in increasing bucket
+     * order.
+     */
+    List<Long> getFirstLiveBuckets(long lowerBoundTimestamp, int limit);
```

Comment:
Did we end up deciding that buckets are now independent from shards? Or, will the store itself be scoped to a given shard?

Reply:
Or, will a given timestamp range be applicable for each shard, and so whatever I use needs to do `(List<Long> bucketIdentifiers, int shardCount) -> List<(Shard, BucketId)>

Reply:
(For context: I'm trying to understand if the work I did to parallel request buckets for each shard is even relevant anymore)

Reply:
discussed: this just goes from `Shard -> List<SweepableBucket>`, or `List<Shard> -> List<SweepableBucket>` (or some other fancy return type)